oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/web/authentication/logout/OidcBackchannelLogoutFilter.java

```diff
@@ -0,0 +1,206 @@
+import org.springframework.security.oauth2.client.oidc.authentication.session.OidcProviderSessionRegistry;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.jwt.BadJwtException;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtDecoderFactory;
+import org.springframework.security.web.authentication.logout.BackchannelLogoutAuthentication;
+import org.springframework.security.web.authentication.logout.BackchannelLogoutHandler;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.util.Assert;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/**
+ * A filter for the Client-side OIDC Backchannel Logout endpoint
+ *
+ * @author Josh Cummings
+ * @since 6.1
+ * @see <a target="_blank" href=
+ * "https://openid.net/specs/openid-connect-backchannel-1_0.html">OIDC Backchannel Logout
+ * Spec</a>
+ */
+public class OidcBackchannelLogoutFilter extends OncePerRequestFilter {
+
+	private final Log logger = LogFactory.getLog(getClass());
+
+	private static final String ERROR_MESSAGE = "{ \"error\" : \"%s\", \"error_description\" : \"%s\" }";
+
+	private final ClientRegistrationRepository clientRegistrationRepository;
+
+	private final JwtDecoderFactory<ClientRegistration> logoutTokenDecoderFactory;
+
+	private RequestMatcher requestMatcher = new AntPathRequestMatcher("/oauth2/{registrationId}/logout", "POST");
+
+	private OidcProviderSessionRegistry providerSessionRegistry = new InMemoryOidcProviderSessionRegistry();
+
+	private LogoutHandler logoutHandler = new BackchannelLogoutHandler();
+
+	/**
+	 * Construct an {@link OidcBackchannelLogoutFilter}
+	 * @param clients the {@link ClientRegistrationRepository} for deriving Logout Token
+	 * validation
+	 * @param logoutTokenDecoderFactory the {@link JwtDecoderFactory} for constructing
+	 * Logout Token validators
+	 */
+	public OidcBackchannelLogoutFilter(ClientRegistrationRepository clients,
+			JwtDecoderFactory<ClientRegistration> logoutTokenDecoderFactory) {
+		this.clientRegistrationRepository = clients;
+		this.logoutTokenDecoderFactory = logoutTokenDecoderFactory;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+			throws ServletException, IOException {
+		RequestMatcher.MatchResult result = this.requestMatcher.matcher(request);
+		if (!result.isMatch()) {
+			chain.doFilter(request, response);
+			return;
+		}
+		String logoutToken = request.getParameter("logout_token");
+		if (logoutToken == null) {
```

Comment:
Would it be logical to move this check after the `if (registration == null)` check? So that firstly, it is checked if URL is correct about registrationId; if not then HTTP 404 is returned. Secondly, it is checked if `logout_token` parameter exists; if it's missing then HTTP 400 should be returned (instead of HTTP 404, which could be confusing). If `logout_token` parameter is missing, returing HTTP 400 would better coform to [OIDC specification](https://openid.net/specs/openid-connect-backchannel-1_0.html#BCResponse) (_If the logout request was invalid [...], the RP MUST respond with HTTP 400 Bad Request._).